dags/analyzer_helper/discourse/transform_raw_info.py

```diff
@@ -0,0 +1,93 @@
+from analyzer_helper.discourse.utils.convert_date_time_formats import DateTimeFormatConverter
+
+class TransformRawInfo:
+
+    def __init__(self):
+        self.converter = DateTimeFormatConverter()
+    
+    def create_data_entry(self, raw_data: list, interaction_type: str = None, interaction_user: int = None):
+        metadata = {
+            "channel_id": raw_data.get("category_id"),
+            "thread_id": raw_data.get("topic_id"),
+            "bot_activity": False,
+        }
+
+        if interaction_type == "reaction":
+            return {
+                "actions": [],
+                "author_id": interaction_user,
+                "date": self.converter.convert_to_datetime(raw_data.get("created_at")),
+                "interactions": [
+                    {
+                        "name": "reaction",
+                        "type": "emitter",  
+                        "users_engaged_id": [str(raw_data["post_id"])],
+                    }
+                ],
+                "metadata": metadata,
+                "source_id": str(raw_data["post_id"]),
+            }
+        else:
```

Comment:
Are we handling all other interaction types in the else condition? Please make sure your code is better formatted. You could either remove the first condition or split each interaction in each if else statement.

Reply:
@polux0